packages/firebase_in_app_messaging/example/android/app/src/main/java/com/example/firebase_in_app_messaging_example/MainActivity.java

```diff
@@ -1,13 +1,22 @@
 package com.example.firebase_in_app_messaging_example;
 
-import android.os.Bundle;
-import io.flutter.app.FlutterActivity;
-import io.flutter.plugins.GeneratedPluginRegistrant;
+import com.example.firebase_in_app_messaging.FirebaseInAppMessagingPlugin;
+import dev.flutter.plugins.e2e.E2EPlugin;
+import io.flutter.embedding.android.FlutterActivity;
+import io.flutter.embedding.engine.FlutterEngine;
+import io.flutter.embedding.engine.plugins.shim.ShimPluginRegistry;
+import io.flutter.plugins.firebaseanalytics.FirebaseAnalyticsPlugin;
 
 public class MainActivity extends FlutterActivity {
+  // TODO(<github-username>): Remove this once v2 of GeneratedPluginRegistrant rolls to stable.
+  // https://github.com/flutter/flutter/issues/42694
   @Override
-  protected void onCreate(Bundle savedInstanceState) {
-    super.onCreate(savedInstanceState);
-    GeneratedPluginRegistrant.registerWith(this);
+  public void configureFlutterEngine(FlutterEngine flutterEngine) {
+    flutterEngine.getPlugins().add(new E2EPlugin());
+    flutterEngine.getPlugins().add(new FirebaseInAppMessagingPlugin());
+    final ShimPluginRegistry shimPluginRegistry = new ShimPluginRegistry(flutterEngine);
+    FirebaseAnalyticsPlugin.registerWith(
```

Comment:
I think firebase analytics is migrated already. Do we need the shim here?

Reply:
Yep, didn't hit stable yet.

Reply:
I'm not sure I understood. The flutterfire repo only has one master branch no?

Reply:
Hmm you may be right considering where we are today.  I don't think I would have done it unless it was necessary.  Honestly this PR has been open so long it's hard to keep track.  I would like to land this because I know it works and we can reevaluate.  If we keep this open I'm sure things will continue to get out of date.